ledger/migration/src/lib.rs

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Ledger migration: Perform updates of LedgerDB to accommodate for
+//! backward-incompatible changes.
+
+use lmdb::{DatabaseFlags, Environment, Transaction, WriteFlags};
+use mc_common::logger::{log, Logger};
+use mc_ledger_db::{
+    key_bytes_to_u64, tx_out_store::TX_OUT_INDEX_BY_PUBLIC_KEY_DB_NAME, u64_to_key_bytes, Error,
+    LedgerDbMetadataStoreSettings, MetadataStore, TxOutStore, TxOutsByBlockValue,
+    BLOCK_NUMBER_BY_TX_OUT_INDEX, COUNTS_DB_NAME, NUM_BLOCKS_KEY, TX_OUTS_BY_BLOCK_DB_NAME,
+};
+use mc_util_lmdb::MetadataStoreError;
+use mc_util_serial::decode;
+use std::path::Path;
+
+const MAX_LMDB_FILE_SIZE: usize = 1_099_511_627_776; // 1 TB
```

Comment:
I notice `mc_leger_db` has a similar `const`; how about we reuse that?

Optional/Equivalent: Use powers of 2 or multiples of 1024.
```suggestion
const MAX_LMDB_FILE_SIZE: usize = 2usize.pow(40); // 1 TB
```